src/calliope/model.py

```diff
@@ -324,11 +322,20 @@ def build(self, force: bool = False, **kwargs) -> None:
             )
         else:
             backend_input = self._model_data
+
+        init_math_list = [] if backend_config.get("ignore_mode_math") else [mode]
+        end_math_list = [] if add_math_dict is None else [add_math_dict]
+        full_math_list = init_math_list + backend_config["add_math"] + end_math_list
+        LOGGER.debug(f"Math preprocessing | Loading math: {full_math_list}")
+        model_math = preprocess.CalliopeMath(full_math_list, self._def_path)
```

Comment:
This "math mode salad" is quite hard to follow because of its "mixed value" nature.
- `init_math_list` will have either `[]` or `[str]`
- the middle (`backend_config["add_math"]`) will have `[]` `[str]`
- `end_math_list` (`add_math_dict`) will have either `None` or `[dict]` 

How about:
```python
math_list = [backend_config.get("mode"), backend_config.get("add_math"), add_math_dict]
math_list = list(filter(bool, math_files))  # Removes False equivalents None, "", [], {}
model_math = preprocess.CalliopeMath(math_list, self._def_path)
```